x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java

```diff
@@ -0,0 +1,316 @@
+                        )
+                    )
+                );
+            }
+        }
+
+        @Override
+        public void close() {
+            if (pkcJwtSignatureValidator != null) {
+                pkcJwtSignatureValidator.close();
+            }
+        }
+
+        @Override
+        public boolean hasUsableJwksAlgs() {
+            return (hmacJwtSignatureValidator != null && hmacJwtSignatureValidator.hasUsableJwksAlgs())
+                || (pkcJwtSignatureValidator != null && pkcJwtSignatureValidator.hasUsableJwksAlgs());
+        }
+
+        private static void validateJwkSettings(
+            RealmConfig realmConfig,
+            boolean isConfiguredJwkSetPkc,
+            boolean isConfiguredJwkSetHmac,
+            boolean isConfiguredJwkOidcHmac
+        ) {
+            if (isConfiguredJwkSetPkc == false && isConfiguredJwkSetHmac == false && isConfiguredJwkOidcHmac == false) {
+                throw new SettingsException(
+                    "At least one of ["
+                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.HMAC_KEY)
+                        + "] or ["
+                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.HMAC_JWKSET)
+                        + "] or ["
+                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.PKC_JWKSET_PATH)
+                        + "] must be set"
+                );
+            }
+
+            // HMAC Key vs HMAC JWKSet settings must be mutually exclusive
+            if (isConfiguredJwkSetHmac && isConfiguredJwkOidcHmac) {
+                throw new SettingsException(
+                    "Settings ["
+                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.HMAC_JWKSET)
+                        + "] and ["
+                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.HMAC_KEY)
+                        + "] are not allowed at the same time."
+                );
+            }
+        }
+
+        // Package private for testing Only
+        Tuple<JwkSetLoader.JwksAlgs, JwkSetLoader.JwksAlgs> getAllJwksAlgs() {
+            final JwkSetLoader.JwksAlgs jwksAlgsHmac;
+            if (hmacJwtSignatureValidator == null) {
+                jwksAlgsHmac = new JwkSetLoader.JwksAlgs(List.of(), List.of());
+            } else {
+                jwksAlgsHmac = ((HmacJwtSignatureValidator) hmacJwtSignatureValidator).jwksAlgs;
+            }
+
+            final JwkSetLoader.JwksAlgs jwksAlgsPkc;
+            if (pkcJwtSignatureValidator == null) {
+                jwksAlgsPkc = new JwkSetLoader.JwksAlgs(List.of(), List.of());
+            } else {
+                jwksAlgsPkc = ((PkcJwtSignatureValidator) pkcJwtSignatureValidator).jwkSetLoader.getContentAndJwksAlgs().jwksAlgs();
+            }
+            return new Tuple<>(jwksAlgsHmac, jwksAlgsPkc);
+        }
```

Comment:
This method exists solely for tests. It is needed so that we can keep existing tests running without major overhaul. We can revisit it later. For now, I am leaving it for convenience.